R/get_current_providers.R

```diff
@@ -155,6 +158,23 @@ use_providers <- function(custom_providers = NULL) {
     }
   }
 
-  options(leaflet.providers = custom_providers)
-  invisible(TRUE)
+  loaded_providers_env$providers <- custom_providers
```

Comment:
Let's maybe call this `provider_obj` to avoid mental name collision with `providers`